File: onlinejudge/leetcode/14-Longest Common Prefix.go
func longestCommonPrefix(strs []string) string {
		sort.Strings(strs)

		firstText := strs[0]
		lastText := strs[len(strs)-1]

		idx := 0
		for idx < len(firstText) && idx < len(lastText) {
				if firstText[idx] == lastText[idx] {
						idx++
				} else {
						break
				}
		}

		return firstText[:idx]
}

File: onlinejudge/leetcode/14-Longest Common Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        
        String firstText = strs[0];
        String lastText = strs[strs.length -1];
        
        int idx = 0;
        while (idx < firstText.length() && idx < lastText.length()) {
            if (firstText.charAt(idx) == lastText.charAt(idx)) {
                idx++;
            } else {
                break;
            }
        }
                
        return firstText.substring(0, idx);
    }
}